backend/app/mcp_server/tools/knowledge.py

```
@@ -0,0 +1,770 @@
+        file_extension: File extension provided by caller (may include leading dot)
+
+    Returns:
+        Normalized file extension without leading dot (e.g., "txt")
+    """
+    ext = (file_extension or "").strip()
+    ext = ext.lstrip(".")
+    if not ext:
+        return DEFAULT_TEXT_FILE_EXTENSION
+    # Basic safety checks (avoid path injection / invalid filenames)
+    if "/" in ext or "\\" in ext or ".." in ext:
+        raise ValueError("Invalid file_extension")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)